### ISSUE TITLE
ssreflect.8.7.dev

### ISSUE THREAD

**Issue by CohenCyril** [coq/opam-coq-archive#208]
Hi @matejkosik @gares ,
I'm surprised to find such a package description. Maybe I missed a discussion? What is it for?
https://github.com/coq/opam-coq-archive/blob/cd8e83337b2c1701a290fd5667e52ddcd4250aa8/extra-dev/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.8.7.dev/opam#L3
Cheers

**Comment by matejkosik** [comment]
This may be my mistake. I suspect that this package is no longer needed.

It made sense in Coq 8.6 but probably does not make sense in Coq 8.7 (or trunk).

**Comment by mattam82** [comment]
So we should just remove it?

**Comment by matejkosik** [comment]
At the moment, `coq-coquelicot.8.7.dev` (which is part of the benchmark suite) depends on `coq-mathcomp-ssreflect.8.7.dev`.

If that dependency is dropped, the installation of `coq-coquelicot.8.7.dev` package fails:
```
mkosik@o1:~$ opam install -v -j4 coq-coquelicot
The following actions will be performed:
  ∗  install coq-coquelicot 8.7.dev

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[coq-coquelicot] https://scm.gforge.inria.fr/anonscm/git/coquelicot/coquelicot.git updated

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[coq-coquelicot: ./autogen.sh] Command started
+ ./autogen.sh  (CWD=/home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev)
- aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
- configure.in:5: warning: prefer named diversions
- configure.in:5: warning: prefer named diversions
- configure.in:5: warning: prefer named diversions
[coq-coquelicot: ./configure] Command started
+ ./configure  (CWD=/home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev)
- checking for g++... g++
- checking whether the C++ compiler works... yes
- checking for C++ compiler default output file name... a.out
- checking for suffix of executables... 
- 
- checking whether we are cross compiling... no
- checking for suffix of object files... o
- checking whether we are using the GNU C++ compiler... yes
- checking whether g++ accepts -g... yes
- checking for coqc... /home/mkosik/.opam/4.03.0/bin/coqc
- checking for coqdep... /home/mkosik/.opam/4.03.0/bin/coqdep
- checking for coqdoc... /home/mkosik/.opam/4.03.0/bin/coqdoc
- checking for SSReflect... no
- File "./conftest.v", line 1, characters 15-43:
- Error:
- Cannot find a physical path bound to logical path matching suffix
- mathcomp.ssreflect.
- 
- configure: error:  *** Unable to find library mathcomp.ssreflect >= 1.6 (http://math-comp.github.io/math-comp/)
[ERROR] The compilation of coq-coquelicot failed at "./configure".
[coq-coquelicot: rm] Command started
+ rm "-r" "-f" "/home/mkosik/.opam/4.03.0/lib/coq/user-contrib/Coquelicot" (CWD=/home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev)

#=== ERROR while installing coq-coquelicot.8.7.dev ============================#
# opam-version 1.2.2
# os           linux
# command      ./configure
# path         /home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev
# compiler     4.03.0
# exit-code    1
# env-file     /home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev/coq-coquelicot-8451-e7966f.env
# stdout-file  /home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev/coq-coquelicot-8451-e7966f.out
# stderr-file  /home/mkosik/.opam/4.03.0/build/coq-coquelicot.8.7.dev/coq-coquelicot-8451-e7966f.err
### stdout ###
# [...]
# checking for C++ compiler default output file name... a.out
# checking for suffix of executables... 
# checking whether we are cross compiling... no
# checking for suffix of object files... o
# checking whether we are using the GNU C++ compiler... yes
# checking whether g++ accepts -g... yes
# checking for coqc... /home/mkosik/.opam/4.03.0/bin/coqc
# checking for coqdep... /home/mkosik/.opam/4.03.0/bin/coqdep
# checking for coqdoc... /home/mkosik/.opam/4.03.0/bin/coqdoc
# checking for SSReflect... no
### stderr ###
# File "./conftest.v", line 1, characters 15-43:
# Error:
# Cannot find a physical path bound to logical path matching suffix
# mathcomp.ssreflect.
# 
# configure: error:  *** Unable to find library mathcomp.ssreflect >= 1.6 (http://math-comp.github.io/math-comp/)
=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
```
So I guess that first we have to fix packages that depend on either `coq-mathcomp-ssreflect.8.7.dev` or on `coq-mathcomp-ssreflect.dev`.

I am not sure how exactly the installation instructions of `coq-coquelicot` must be changed so that installation goes through even if `coq-mathcomp-ssreflect` is not installed.

**Comment by mattam82** [comment]
@silene might tell us what’s the best course of action here
Le ven. 1 déc. 2017 à 14:03, Matej Košík <notifications@github.com> a
écrit :

**Comment by silene** [comment]
> silene might tell us what’s the best course of action here

There is nothing special here: Coquelicot depends on Ssreflect. Also note that I am not the maintainer of the `coq-coquelicot.8.7.dev` package. The latest released version of Coquelicot is 3.0.

**Comment by CohenCyril** [comment]
@silene what is surprising here is the presence of the package itself while official versions of ssreflect and mathematical components are coq-mathcomp-* with versions of the form 1.*...
I think there should be a package named coquelicot.3.0 and that the packages ssreflect.8.7.dev and coq-coquelicot.8.7.dev should be removed to avoid confusion for end users.

**Comment by mattam82** [comment]
@CohenCyril the extra-dev repository is specifically about development versions of packages, so I don't see why you're surprised.

**Comment by mattam82** [comment]
We can surely remove the coquelicot and ssreflect dev packages from the extra-dev repos if you guys want.

**Comment by CohenCyril** [comment]
The development version of ssreflect is here: https://github.com/coq/opam-coq-archive/tree/master/extra-dev/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.dev with the correct dependencies.

**Comment by mattam82** [comment]
@CohenCyril I see. We have to remove theses 8.7.dev packages from extra-dev but we could move these to a benchmark-specific repo if necessary. @matejkosik why not just benchmark coq-mathcomp-ssreflect.dev instead of our own package?

**Comment by matejkosik** [comment]
@mattam82 

> why not just benchmark coq-mathcomp-ssreflect.dev instead of our own package?

I have no idea.

I don't follow the benchmarking story. Is that machinery still relevant? Is it worth to invest more time to it @maximedenes , @ejgallego or is scheduled to be replaced with something superior?

(If yes, then I would refrain from making any changes --- partially because I can't actually access `pendulum` machine and update/test/maintain the script --- partially I am not keen to do work that would later be thrown away anyway).

**Comment by CohenCyril** [comment]
Fixed by #258 (not closed automatically...)
